src/main/java/http_server/Router.java

```diff
@@ -25,6 +25,17 @@
     private static String dirPath;
 
     public Router() {
+        initializeRouter();
+    }
+
+    public Router(String staticDirectoryRelativePath) {
+        initializeRouter();
```

Comment:
Since the `Router()` constructor only calls `initializeRouter`, and this constructor does some extra work, you should look into constructor chaining - you can call one constructor from another constructor.